conventions.md

```diff
@@ -0,0 +1,210 @@
+  - Always load environment variables from a `.env` or `config.sh` file and configuration parameters from a dedicated config file.  
+  - Remember to update `.env.example` or `config.example.sh` accordingly.  
+  - Declare global variables (like `CONTRACT_DIRECTORY`, `DEPLOY_SCRIPT_DIRECTORY`, etc.) in the `.env` or `config.sh` files so they are consistently available across all functions.
+- **Error handling and logging:**  
+  - Use dedicated helper functions for logging (e.g., `echoDebug`, `error`, `warning`, `success`) to provide consistent, color-coded feedback.  
+  - Validate external inputs and environment variables early in the script, and exit with a clear error message if required variables are missing.  
+  - **Child function failure checking:** When calling any underlying child function, immediately check its exit status using the `checkFailure` helper function. This ensures that if a function fails, the script halts further execution, preventing cascading errors.  
+  - If installing a new system package, add it to `preinstall.sh` so that developers have a unified initial installation procedure.
+- **User interaction and prompts:**  
+  - When user input is needed (e.g., selecting a network or confirming an action), use clear prompts with descriptive instructions. Tools like `gum choose` can enhance usability.  
+  - Document any TODOs and known limitations clearly at the beginning of the script for future improvements.
+- **Code readability and maintenance:**  
+  - Use consistent indentation, naming conventions, and comments throughout the script.  
+  - Provide usage instructions at the top of each script if applicable.  
+  - Separate core operations (e.g., deploying contracts, updating logs) into distinct, well-documented functions to facilitate easier maintenance and testing.
+
+# Audit logs and reports conventions
+
+- **Audit log file (`auditLog.json`):**  
+  Contains two main sections:
+  1. **audits:** Each audit entry has a unique ID (e.g., `auditYYYYMMDD`), an auditCompletedOn date, auditedBy, auditorGitHandle (if applicable), auditReportPath (PDF location in `audit/reports/`), and an auditCommitHash.
+  2. **auditedContracts:** Maps contract names and versions to the relevant audit IDs in the audits section.
+- **Storing reports:**  
+  Place PDF reports in the `audit/reports/` directory, using a naming format that includes the date and contract info (e.g., `2025.01.17_LiFiDexAggregator(v1.3.0).pdf`).
+- **Adding new audits:**  
+  For new or updated contracts, add an entry under audits with the correct date, auditReportPath, and auditCommitHash. Then, reference that new ID in auditedContracts for each relevant version.
+- **Naming and versioning:**  
+  Use a format like `auditYYYYMMDD` for unique IDs and ensure that contract versioning is consistent between the code and the `auditLog.json`.
+- **Essential fields:**  
+  - `auditCompletedOn`: Date in `DD.MM.YYYY` or `YYYY-MM-DD` format.  
```

Comment:
I like this bullet list more than line 191 that names all of them in text form....maybe remove the above?